build.rs

```diff
@@ -8,16 +8,18 @@ use std::process::Command;
 
 fn main() {
     // Read environment variables set by cargo
-    let out_dir_path = env::var("OUT_DIR").expect("Missing OUT_DIR environment variable");
-    let out_dir = Path::new(&out_dir_path);
-
     let cargo_path = env::var("CARGO").expect("Missing CARGO environment variable");
     let cargo = Path::new(&cargo_path);
 
     let manifest_dir_path =
         env::var("CARGO_MANIFEST_DIR").expect("Missing CARGO_MANIFEST_DIR environment variable");
     let manifest_dir = Path::new(&manifest_dir_path);
 
+    // Find the root project target dir
+    let current_dir = env::current_dir().expect("Couldn't get current directory");
+    let target_dir_rel = manifest_dir.join("target");
```

Comment:
For crates.io dependencies I believe the current dir is the source dir of the dependency, not the source dir of the original project.

Reply:
Ping @rybot666?